src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,199 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String MIN_LENGTH_VALID = "6chars";
+    private static final String MIN_EDGE_LENGTH_VALID = "7 chars";
+    private static final String MAX_EDGE_LENGTH_INVALID = "5char";
+    private static final String IS_EMPTY = "";
+    private static final String IS_BLANK = "      ";
+    private static final String VALID_PASSWORD = "ValidPassword";
+    private static final int VALID_AGE = 19;
+    private static final int MIN_VALID_AGE = 18;
+    private static final int MAX_INVALID_AGE = 17;
+    private static final int AGE_IS_ZERO = 0;
+    private static final int AGE_IS_NEGATIVE = -18;

```

Comment:
I consider constants in tests a bad pattern

simply put - it makes your test methods less readable, you always need to scroll to find a corresponding value
apart from that, it makes your test method depend on a single constant, which gets even worse, if you use that constant in several places

while constants are ok for production code, they're not ok for test code :)

I recommend reading this [article](https://medium.com/@erik.sacre/on-tdd-literals-versus-constants-in-tests-8ac992c7873f)